jax/experimental/jax2tf/tfjs_demo/Sketcher/utils.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from jax import numpy as jnp # type: ignore
+from jax.experimental import jax2tf # type: ignore
+jax2tf.jax2tf.ENABLE_TF_CONVOLUTION = True
+
+import numpy as np # type: ignore
+import os
+import requests # type: ignore
+import tensorflow as tf # type: ignore
+from typing import Callable, Dict, List, Sequence
+
+class ExportWrapper(tf.train.Checkpoint):
+
+  def __init__(self, fn: Callable, params,
+               input_shape: Sequence[int],
+               batch_sizes: Sequence[int]):
+    """
+    Args:
+      fn: a function taking two arguments, the parameters and the batch of
+        images.
+      params: the parameters, as a list/tuple/dictionary of np.ndarray, to be
+        used as first argument for `fn`.
+     input_shape: the shape of the second argument of `fn` (except the batch size)
+     batch_sizes: a sequence of batch sizes for which to save the function.
+    """
+    super().__init__()
+    # Convert fn from JAX to TF.
+    # with_gradient must be set to True to be able to convert the saved model
+    # to TF.js, as "PreventGradient" is not supported.
+    with_gradient = True
+    self._fn = jax2tf.convert(fn, with_gradient=with_gradient)
+    # Create tf.Variables for the parameters.
+    self._params = tf.nest.map_structure(
+        # If with_gradient=False, we mark the variables behind as non-trainable,
+        # or else the Keras model below fails for trying to access them
+        # (even with hub.KerasLayer(..., trainable=False), surprisingly).
+        lambda param: tf.Variable(param, trainable=with_gradient),
+        params)
+    self._signatures: Dict[str, Callable] = {}
+
```

Comment:
@gnecula this was part of the `ExportWrapper`, but I don't see it mentioned in the spec. Is it needed?

Reply:
I do not think so.

Reply:
Thank you, I removed it.